src/core/mount.c

```diff
@@ -1729,7 +1729,9 @@ static int mount_load_proc_self_mountinfo(Manager *m, bool set_flags) {
 
         assert(m);
 
-        r = libmount_parse(NULL, NULL, &table, &iter);
+        r = libmount_parse_fsinfo(&table, &iter);
```

Comment:
> hmm, why is this a new function? can't this be hidden behind the same libmount_parse() call?

Ok, having looked at this I remember now how this evolved.

I started out to do what you suggested but the overhead of getting the entire mount table and all mount attributes at once wasn't any better than we have reading the entire proc file.

I suggested getting the mount attributes on demand to get a little more out of the fsinfo use case and Karel like that idea a lot and based his implementation on it.

The catch then is that library uses need to check for valid returns when using the mount info. get calls hence using the fsinfo implementation needs to be a conscious decision made by the library user to ensure that validity checks are made when using retrieved fields.

As you can see in the cover letter I mention using just this fsinfo based table parsing doesn't get much improvement.
Also from the cover letter you can see the real benefit of this change is the use of the mount notifications together with fsinfo.

So we could just drop this part of the change since it doesn't get us much actual improvement and the notifications changes don't really need it either.

Reply:
The other option would be to merge this into the existing libmount_parse() function and ensure the mount attribute get  values are checked. The check is a good idea anyway, along with comments about the need for them.

I admit I'm pretty sure I also haven't been completely diligent about checking "all" the returned values either and I also need to go back to libmount and make sure there are clear invalid returns for all the mount attribute get functions too.